MergerService/src/run.cs

```diff
@@ -299,10 +293,7 @@ private void RunTask(MergeTask task, ITaskUtils taskUtils)
 
                 try
                 {
-                    task.Status = Status.FAILED;
-                    task.Reason = e.Message;
-                    UpdateParameters updateParameters = task.GetUpdateParameters();
-                    taskUtils.UpdateTask(task.JobId, task.Id, updateParameters);
+                    taskUtils.UpdateFailed(task.JobId, task.Id, e.Message);
```

Comment:
should use reject function.
the update failed don't update the resettable and attempts

Reply:
Changed.